module/util/log.go

```diff
@@ -0,0 +1,20 @@
+package util
+
+import "github.com/rs/zerolog"
+
+func LogProgress(msg string, total int, logger *zerolog.Logger) func(current int) {
+	lookup := make(map[int]int)
+	// report every 10 percent
+	for i := 1; i < 10; i++ { // [1...9]
+		lookup[total/10*i] = i * 10
+	}
+	if total > 0 {
+		lookup[total-1] = 100
+	}
+	return func(current int) {
+		percentage, ok := lookup[current]
```

Comment:
if current is for whatever reason greater than total, than this stops logging.  instead of using a lookup map, maybe just do this

```
func LogProgress(msg string, total int, logger *zerolog.Logger) func(current int) {
	logThreshold := float64(0)
	return func(current int) {
		percentage := float64(100)
		if total > 0 {
			percentage = (float64(current+1) / float64(total)) * 100. // current+1 assuming zero based indexing 
		}

		if percentage >= logThreshold+10 {
			logger.Info().Msgf("%s completion percentage: %v percent", msg, percentage)
			logThreshold += 10
		}
	}
}
```